src/panels/lovelace/hui-view.js

```diff
@@ -76,36 +76,13 @@ class HUIView extends PolymerElement {
   constructor() {
     super();
     this._elements = [];
-    this._whenDefined = {};
   }
 
   _getElements(cards) {
     const elements = [];
 
     for (let i = 0; i < cards.length; i++) {
-      let error = null;
-      let cardConfig = cards[i];
-      let tag;
-      if (!cardConfig.type) {
-        error = 'Card type not configured.';
-      } else {
-        tag = computeCardElement(cardConfig.type);
-        if (tag === null) {
-          error = `Unknown card type encountered: "${cardConfig.type}".`;
-        } else if (!customElements.get(tag)) {
-          error = `Custom element doesn't exist: "${tag}".`;
-          if (!(tag in this._whenDefined)) {
-            this._whenDefined[tag] = customElements.whenDefined(tag)
```

Comment:
It is important that we don't lose this logic. Without this, custom cards are too fragile to be hurt by race conditions. Add [the custom card example](https://developers.home-assistant.io/docs/en/lovelace_custom_card.html) to your config and reload the page. You'll see that it won't be done in time and now will never show up.

I can think of 2 solutions. First one is to allow passing in a callback to `createCardElement` that can be called to trigger a refresh. But what I think is better is to just fire an event when the card is defined and have `hui-root` listen to it and rebuild the UI. 

To fire an event from a function, you can extract the function from the [EventsMixin](https://github.com/home-assistant/home-assistant-polymer/blob/master/src/mixins/events-mixin.js#L50-L62) into a util function.

Reply:
We have a node, the one the function created :)